src/providers/Global/Global.provider.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useReducer, useContext } from 'react';
+
+const GlobalContext = React.createContext();
+
+export const ACTIONS = {
+  TOOGLE_DARK: 'toogleDark',
+  UPDATE_SEARCH: 'updateSearch',
+};
+
+const INITIAL_STATE = {
+  isDarkMode: false,
+  searchTerm: '',
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case ACTIONS.TOOGLE_DARK:
+      return {
+        ...state,
+        isDarkMode: !state.isDarkMode,
+      };
+    case ACTIONS.UPDATE_SEARCH:
+      return {
+        ...state,
+        searchTerm: action.payload,
+      };
+    default:
+      throw new Error(`Global reducer action.type ${action.type} not allowed`);
+  }
+};
```

Comment:
I suggest you to extract the reducer logic to its own file, so you can test it isolated.